Core/PARStore.m

```diff
@@ -949,6 +950,85 @@ - (void)setEntriesFromDictionary:(NSDictionary *)dictionary
      }];
 }
 
+- (void)appendChanges:(NSArray *)changes forDeviceIdentifier:(NSString *)deviceIdentifier
+{
+    // Model and PSC
+    NSManagedObjectModel *mom = [PARStore managedObjectModel];
+    NSPersistentStoreCoordinator *psc = [[NSPersistentStoreCoordinator alloc] initWithManagedObjectModel:mom];
+    
+    // Persistent Store
+    NSError *error = nil;
+    NSString *dirPath = [self directoryPathForDeviceIdentifier:deviceIdentifier];
+    [[NSFileManager defaultManager] createDirectoryAtPath:dirPath withIntermediateDirectories:NO attributes:nil error:NULL];
+    [self addPersistentStoreWithCoordinator:psc dirPath:dirPath readOnly:NO error:&error];
+    
+    // Context
+    NSManagedObjectContext *moc = [[NSManagedObjectContext alloc] initWithConcurrencyType:NSPrivateQueueConcurrencyType];
+    [moc setPersistentStoreCoordinator:psc];
+    [moc setUndoManager:nil];
+
+    // Have main context observe changes made in background, and merge them.
+    id observer = [[NSNotificationCenter defaultCenter] addObserverForName:NSManagedObjectContextDidSaveNotification object:moc queue:nil usingBlock:^(NSNotification *notification) {
+        [self.databaseQueue dispatchSynchronously:^{
+            [[self managedObjectContext] mergeChangesFromContextDidSaveNotification:notification];
+        }];
+    }];
+    
+    // Add changes
+    [moc performBlockAndWait:^{
+        NSError *error = nil;
+        
+        // Expression to retrieve the maximum timestamp in the store
```

Comment:
I can certainly change this method so that it uses queries. I agree that would be more robust.

My main concern is that given that PARStore assumes append only for each device, I think it may require far reaching changes to PARStore and the revision manager if changes are able to be inserted at any point in the history. (Reloading the whole document history every sync is not an option IMO.)

I will see if I can make the change, but it does not solve the issue of potentially inserting into the history of PARStore and breaking the append-only assumption.

When you ask 'why not get the last row'? A database is not ordered, so I would presumably need to do some fetch anyway, with sorting. That would be slower than the @max function I used. But if you have a better idea, let me know. (BTW I don't want to just access cached timestamps, because this activity is happening on a different thread, and I don't want race conditions. Querying the DB itself is safest.)